### DISCUSSION TITLE
Can't get it to work, running into error?

### DISCUSSION THREAD

**Discussion by ChikoritaTrainer** [Crivella/ocr_translate#29] [Q&A]
So I've been looking for a tool like this to read manga for a while, but when I try to use t the server gives me a bunch of errors. I think I'm doing everything right, but the simplicity makes me think I might've skipped a step somewhere. Maybe it is working but I'm missing something???

Here's the weird thing it's doing:

![Screenshot 2024-05-29 225504](https://github.com/Crivella/ocr_translate/assets/66294377/0e34d755-4da8-414d-b49d-385467103b62)
![Screenshot 2024-05-29 225516](https://github.com/Crivella/ocr_translate/assets/66294377/c8646e07-1856-4cbd-ae52-55f6a1aba0a8)
![Screenshot 2024-05-29 225520](https://github.com/Crivella/ocr_translate/assets/66294377/b1c2420e-66eb-4727-b0f6-d0eb6d99dd5b)

**Comment by Crivella** [comment]
Are you using the `ocr_extension` for firefox or some custom tool you made to talk to the server?

If it is the latter,  you might want to check out how the API is being called in the firefox extension (https://github.com/Crivella/ocr_extension/blob/master/src/utils/API.js).

Regarding the errors the 406 is just the server attempting to get a translation for something that was already translated, the 400 can be a lot of things (see all the parts returning 400 in this function https://github.com/Crivella/ocr_translate/blob/95b26839d9d95103214caee419a33cb7b2bbad9f/ocr_translate/views.py#L242) and you can get more information by the object returned to whatever is making the call.

**Reply by ChikoritaTrainer** [reply]
I'm using the firefox extension. 
Frankly, I'm not all that coding-savvy so the last line is just gibberish for me >.<
I know what returning and objects are but have no idea where or how to check or even what thing to check for
Where do I start trying to fix it?

**Reply by ChikoritaTrainer** [reply]
Here's a more compact image of the whole process from me loading less images at once

![image](https://github.com/Crivella/ocr_translate/assets/66294377/b3f38b3b-5b17-4825-bf98-065844f4f037)

**Comment by Crivella** [comment]
Are you loading the model you want to use from the extension? I do not see any message in the log that would indicate so, eg if i try to activate the extension without models loaded i get the same sequence of 406 and400 errors
![image](https://github.com/Crivella/ocr_translate/assets/34096612/5788e7db-d568-41bb-bb71-bc0cd2dda22b)

When you are loading a model you would expect something like (minus the DEBUG stuff which you won't see unless you activate it yourself by using [the appropriate environment variable](https://crivella.github.io/ocr_translate/user/envs.html))
![image](https://github.com/Crivella/ocr_translate/assets/34096612/a5a4e579-4b02-4e81-ae97-83a2fac20aca)

You activate language first and model after by selecting them in the pop up menu of the extension (ti might be hidden in the puzzle symbols if you have many extensions installed 
![image](https://github.com/Crivella/ocr_translate/assets/34096612/0549f81e-89e2-4ff0-a1ac-cf7ade424033)
)
![image](https://github.com/Crivella/ocr_translate/assets/34096612/1a29957a-9aa1-4fc0-8380-23df73fd7771)
(The stuff menus turn green once you click submit and the server respond with a success)

**Reply by ChikoritaTrainer** [reply]
Ah, so it seems I did miss a step! I completely just did not see the pop-up menu! 

It's working just fine now!

Thank you very much :)

**Reply by Crivella** [reply]
Np happy to help enjoy the translations ;P
